### ISSUE TITLE
Anything new?

### ISSUE THREAD

**Issue by DarknesGaming** [j-selby/EasyJetpack#4]
I just noticed that you updated the license for the project, and that you merged my work. Anything big happening with EasyJetpack soon?

**Comment by j-selby** [comment]
The only reason I put the license there was because there was ambiguity about the licensing of the project, but I never actually pushed it (it has been listed as GPLv3 on BukkitDev since it's launch).

I have no real plans for the project at this point in time, as I have moved on to emulators, as well as some more computer science-y projects. I might come back to Bukkit at some point, but I see no real point as I don't really see much of a MC server community nowadays (I could be wrong though), and it would likely end up as wasted effort. The recent activity on forks of this project might prove me wrong however.

The programming for EasyJetpack was, at the very least, horrible, and at worst, server crashing. Looking back on it now doesn't really make me grin. I'll see though.

**Comment by DarknesGaming** [comment]
The programming isn't bad. At least Paul Vogel managed to make your plugin compatible with Minecraft 1.9 and above.
I haven't noticed any kind of crashing whatsoever, and your plugin still works. You should merge with Paul Vogel's work, compile the plugin, and publish that to Bukkit. It's hard to find plugins that involve jetpacks, even on Curse and SpigotMC.
Speaking of SpigotMC, the MC server community is still strong. People are choosing Spigot over Bukkit nowadays, mostly because that means a high-performance, and lag-free server for them.
If you really want to see how great your plugin is, you can try going on Epiccraft. That server uses your plugin, and it has been for a year.
IP: epiccraft-mc.no-ip.org

**Comment by DarknesGaming** [comment]
I managed to allow your plugin to hook to NoCheatPlus's exemption manager.
Paul Vogel's work should allow your plugin to work with Minecraft 1.9 and beyond. Plus, it will give you a reason to make your plugin popular again.

**Comment by j-selby** [comment]
I've started a new version of the plugin in https://github.com/j-selby/EasyJetpack/tree/2.0.0 - this should be much cleaner then before, and should handle situations better. Still not feature complete (its a total recode), but it should do the job, and Jetpacks can be defined now via the configuration file itself.

**Comment by DarknesGaming** [comment]
You are right. The code is much cleaner, but there's no longer an API. This could take me a long time to make the old code work with this new one. I should wait until version 2.0.0 is complete.

**Comment by j-selby** [comment]
Yes, strictly there is no longer an API, however, jetpacks can be entirely added via the configuration - just leave out what components you don't want, or change (so, you can add craftable items without actions, for example). An API is a bit far off, but I'm more thinking about a custom action provider or similar.

**Comment by DarknesGaming** [comment]
With this code, that means I would have a hard time making jetpacks I make use a different class file with it's own set of visual effects.
Example: I want to make a jetpack with colored dust and not smoke. I can make a new class file, but I have no way of making my jetpacks use that instead of the normal smoke. How can I do this?

**Comment by j-selby** [comment]
Yeah, I understand that. Right now, I'm trying to perfect the stock implementation itself, as the configuration options offered through this new mechanism will be plenty for most users. Custom Jetpacks should be able to be added very shortly via the JetpackManager class, as soon as Jetpacks themselves can be instantiated independently from the configuration. This would be achieved through another (extension) plugin having a dependency on EasyJetpack and using the aforementioned JetpackManager.

This is just the plan as it is, but I'm welcome to suggestions in this regard.
